### PR TITLE
*/*: fix building with gcc-10

### PR DESCRIPTION
**Pull request by jsmolic**

### CODE REVIEW

File: x11-wm/awesome/awesome-4.3-r1.ebuild [hunk 1]
```diff
@@ -54,13 +53,15 @@ DEPEND="${RDEPEND}
 		dev-lua/luacheck
 	)
```
Comment (SoapGentoo) [inline x11-wm/awesome/awesome-4.3-r1.ebuild:54]: can you check the build system here? My gut feeling is that

- app-text/asciidoc
- app-text/xmlto
- dev-util/gperf
- media-gfx/imagemagick[png]
- doc? ( dev-lua/ldoc )
- and app-shells/zsh in test

can likely also go into `BDEPEND`

File: x11-wm/awesome/awesome-4.3-r1.ebuild [hunk 2]
```diff
@@ -54,13 +53,15 @@ DEPEND="${RDEPEND}
...
 # Skip installation of README.md by einstalldocs, which leads to broken symlink
 DOCS=()
 PATCHES=(
-	"${FILESDIR}/${PN}-4.0-convert-path.patch"  # bug #408025
-	"${FILESDIR}/${PN}-xsession.patch"          # bug #408025
-	"${FILESDIR}/${PN}-4.0-cflag-cleanup.patch" # bug #509658
+	"${FILESDIR}"/${PN}-4.0-convert-path.patch  # bug #408025
+	"${FILESDIR}"/${PN}-xsession.patch          # bug #408025
+	"${FILESDIR}"/${PN}-4.0-cflag-cleanup.patch # bug #509658
+	"${FILESDIR}"/${P}-fno-common.patch			# bug #707262
```
Comment (SoapGentoo) [inline x11-wm/awesome/awesome-4.3-r1.ebuild:64]: use spaces?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: app-misc/utimer/utimer-0.4-r1.ebuild [hunk 3]
```diff
@@ -12,21 +12,18 @@ SLOT="0"
...
-DOCS=( AUTHORS ChangeLog NEWS README )
+PATCHES=(
+	"${FILESDIR}"/${P}-fno-common.patch
+	"${FILESDIR}"/${P}-locale.patch
+)
 
 src_configure() {
 	local myconf=( --enable-debug=no )
 	use debug || myconf=( --enable-debug=yes )
 	econf "${myconf[@]}" $(use_enable nls)
```
[the remark this answers is not in the document]
Comment (jsmolic) [reply app-misc/utimer/utimer-0.4-r1.ebuild:28]: changed it locally but forgot to force push :smile:

File: x11-wm/awesome/awesome-4.3-r1.ebuild (hunk 1 above)
Comment (jsmolic) [reply x11-wm/awesome/awesome-4.3-r1.ebuild:54]: It seems off to me as well, will check it

File: app-misc/utimer/utimer-0.4-r1.ebuild (hunk 3 above)
[the remark this answers is not in the document]
Comment (jsmolic) [reply app-misc/utimer/utimer-0.4-r1.ebuild]: Done ;)

File: app-misc/vifm/vifm-0.11.ebuild [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+src_install() {
+	default
+
+	if use vim; then
+		local t
+		for t in app plugin; do
+			insinto /usr/share/vim/vimfiles/"${t}"
+			doins "${S}"/data/vim/doc/"${t}"/"${PN}"*
```
[the remark this answers is not in the document]
Comment (jsmolic) [reply app-misc/vifm/vifm-0.11.ebuild]: Done

File: app-misc/vifm/vifm-0.11.ebuild [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		done
+	fi
+
+	if use vim-syntax; then
+		local t
+		for t in ftdetect ftplugin syntax; do
+			insinto /usr/share/vim/vimfiles/"${t}"
+			doins "${S}"/data/vim/"${t}"/"${PN}".vim
```
[the remark this answers is not in the document]
Comment (jsmolic) [reply app-misc/vifm/vifm-0.11.ebuild]: Done

File: x11-wm/awesome/awesome-4.3-r1.ebuild (hunk 2 above)
Comment (jsmolic) [reply x11-wm/awesome/awesome-4.3-r1.ebuild]: Done

File: x11-wm/awesome/awesome-4.3-r1.ebuild (hunk 1 above)
Comment (jsmolic) [reply x11-wm/awesome/awesome-4.3-r1.ebuild:54]: Reworked it a little bit, have a look. Version specifiers are outdated and always satisfied so I dropped them. Looks like `gperf` and `xmlto` are not used anymore, there might be more incorrect deps

File: x11-wm/awesome/awesome-4.3-r2.ebuild [hunk 6]
```diff
@@ -3,64 +3,70 @@
 
 EAPI=7
 
-inherit cmake-utils desktop pax-utils
+inherit cmake desktop pax-utils
+
+if [[ ${PV} == 9999 ]] ; then
```
Comment (SoapGentoo) [inline x11-wm/awesome/awesome-4.3-r2.ebuild:8]: `if [[ ${PV} == *9999 ]] ; then` (catches live LTS branches)

File: x11-wm/awesome/awesome-4.3-r2.ebuild [hunk 7]
```diff
@@ -76,16 +82,11 @@ src_configure() {
 		mycmakeargs+=("-DLUA_INCLUDE_DIR=${EPREFIX}/usr/include/luajit-2.0")
 		mycmakeargs+=("-DLUA_LIBRARY=${EPREFIX}/usr/$(get_libdir)/libluajit-5.1.so")
 	fi
-	cmake-utils_src_configure
-}
-
-src_test() {
```
Comment (SoapGentoo) [inline x11-wm/awesome/awesome-4.3-r2.ebuild:82]: did you run tests?

File: x11-wm/awesome/awesome-9999.ebuild [hunk 8]
```diff
@@ -5,15 +5,21 @@ EAPI=7
 
 LUA_COMPAT=( lua5-{1..3} luajit )
 
-inherit cmake desktop git-r3 lua-single pax-utils
+inherit cmake desktop lua-single pax-utils
+
+if [[ ${PV} == 9999 ]] ; then
```
Comment (SoapGentoo) [inline x11-wm/awesome/awesome-9999.ebuild:10]: ditto

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: x11-wm/awesome/awesome-4.3-r2.ebuild (hunk 6 above)
Comment (jsmolic) [reply x11-wm/awesome/awesome-4.3-r2.ebuild]: Done

File: x11-wm/awesome/awesome-4.3-r2.ebuild (hunk 7 above)
Comment (jsmolic) [reply x11-wm/awesome/awesome-4.3-r2.ebuild:82]: I tried, but tests fail. They are already restricted, I just added a bug reference and restored `src_test` if someone wants to fix it in the future.

File: x11-wm/awesome/awesome-9999.ebuild (hunk 8 above)
Comment (jsmolic) [reply x11-wm/awesome/awesome-9999.ebuild]: Done

File: sys-block/vblade/files/vblade-24-fno-common.patch [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+--- a/makefile
++++ b/makefile
+@@ -8,8 +8,8 @@ sbindir = ${prefix}/sbin
+ sharedir = ${prefix}/share
+ mandir = ${sharedir}/man
+ 
+-O=aoe.o bpf.o ${PLATFORM}.o ata.o
+-CFLAGS += -Wall -g -O2
++O=aoe.o bpf.o ${PLATFORM}.o ata.o dat.o
++CFLAGS += -Wall -g -O2 -fno-common
```
Comment (SoapGentoo) [inline sys-block/vblade/files/vblade-24-fno-common.patch:58]: this seems wrong, change it to `CFLAGS += -Wall`

File: sys-block/vblade/files/vblade-24-fno-common.patch [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+ CC = gcc
+ 
+ vblade: $O
+@@ -27,6 +27,9 @@ ata.o : ata.c config.h dat.h fns.h makefile
+ bpf.o : bpf.c
+ 	${CC} ${CFLAGS} -c $<
+ 
++dat.o : dat.c
++	${CC} ${CFLAGS} -c $<
```
Comment (SoapGentoo) [inline sys-block/vblade/files/vblade-24-fno-common.patch:67]: add `${CPPFLAGS}` (user flag that should be respected)

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sys-block/vblade/files/vblade-24-fno-common.patch (hunk 9 above)
Comment (jsmolic) [reply sys-block/vblade/files/vblade-24-fno-common.patch]: :+1: Done

File: sys-block/vblade/files/vblade-24-fno-common.patch (hunk 10 above)
Comment (jsmolic) [reply sys-block/vblade/files/vblade-24-fno-common.patch]: Done

**Comment by thesamesam** [comment]
Excellent work! Make sure you keep bug links in patches ideally (or add them in if they don't exist). I follow [this](https://dev.gentoo.org/~vapier/clean-patches) guide roughly (but you were right to strip out that git nonsense from the awesomewm patch I added in a hurry ;)).

**Comment by jsmolic** [comment]
@thesamesam Thanks :)
Ususally I keep patches completely stripped down, something I picked up from @SoapGentoo :smiley:
Thanks for linking the guide, I'll add bug references in the future

**Pull request closed without merging**